rpcclient/daemon.go

```diff
@@ -0,0 +1,28 @@
+package rpcclient
+
+import (
+	"github.com/athanorlabs/atomic-swap/rpc"
+)
+
+// Shutdown swapd
+func (c *Client) Shutdown() error {
+	const (
+		method = "daemon_shutdown"
+	)
+	if err := c.Post(method, nil, nil); err != nil {
+		return nil // Does not expect a response from swapd
```

Comment:
we should probably still return the error here, in case of other errors - I don't think it errors just because no response is expected? correct me if I'm wrong

Reply:
Correct.  There is always an HTTP response to any HTTP request, even if the response body is empty. I could see making an exception for `errors.Is(err, syscall.ECONNREFUSED)` and not treating that as an error, as it may indicate that the server was already shutdown, but I would shy away from that implementation until someone shows their use case of why the behavior is needed.

Reply:
Here's an end-to-end unit test for the shutdown RPC method that can be placed in the `daemon` package:
```
// Tests the shutdown RPC method
func TestRunSwapDaemon_RPC_Shutdown(t *testing.T) {
	conf := createTestConf(t, tests.GetMakerTestKey(t))
	timeout := time.Minute
	ctx := launchDaemons(t, timeout, conf)

	c := rpcclient.NewClient(ctx, fmt.Sprintf("http://127.0.0.1:%d", conf.RPCPort))
	err := c.Shutdown()
	require.NoError(t, err)

	err = c.Shutdown()
	require.ErrorIs(t, err, syscall.ECONNREFUSED)
}
```
The test isn't working yet, so it may need further modifications. In the test helper method `launchDaemons`, you'll also need to change a couple lines to allow `nil` as an allowed return value from `RunSwapDaemon`:
```
			err := RunSwapDaemon(ctx, conf)
			if err != nil {
				require.ErrorIs(t, err, context.Canceled)
			}
```
I can debug it more if you want. It almost looks like the server is shutting down before it sends the RPC response. If that's the case, we'll need to figure out how to trigger a graceful shutdown that finishes serving the request, or switch to the alternative shutdown method of calling cancel on the outermost server context.